packages/common/src/templates/HubTemplate.ts

```diff
@@ -0,0 +1,225 @@
+import { IArcGISContext } from "../ArcGISContext";
+import { getEditorConfig } from "../core/schemas/getEditorConfig";
+import { IEditorConfig } from "../core/behaviors/IWithEditorBehavior";
+import { HubItemEntity } from "../core/HubItemEntity";
+import { IHubTemplate, IHubTemplateEditor } from "../core/types/IHubTemplate";
+import { TemplateEditorType } from "./_internal/TemplateSchema";
+import { DEFAULT_TEMPLATE } from "./defaults";
+import {
+  createTemplate,
+  deleteTemplate,
+  editorToTemplate,
+  updateTemplate,
+} from "./edit";
+import { fetchTemplate } from "./fetch";
+import { IEntityEditorContext } from "../core/types";
+import { enrichEntity } from "../core/enrichEntity";
+import { cloneObject } from "../util";
+
+/** Hub Template Class */
+export class HubTemplate extends HubItemEntity<IHubTemplate> {
+  /**
+   * Private constructor to allow for future
+   * template-specific logic
+   * @param template
+   * @param context
+   */
+  private constructor(template: IHubTemplate, context: IArcGISContext) {
+    super(template, context);
+  }
+
+  /**
+   * Create an HubTemplate instance from an IHubTemplate object
+   * @param json - JSON object to create a HubTemplate from
+   * @param context - ArcGIS context
+   * @returns
+   */
+  static fromJson(
+    json: Partial<IHubTemplate>,
+    context: IArcGISContext
+  ): HubTemplate {
+    // merge what we have with the default values
+    const pojo = this.applyDefaults(json, context);
+    return new HubTemplate(pojo, context);
+  }
+
+  /**
+   * Create a new HubTemplate, returning a HubTemplate instance.
+   * This does not automatically persist the Template into
+   * the backing store unless save is set to true
+   *
+   * NOTE: we have no immediate plans to allow template creation
+   * from the context of our application, but scaffolding this
+   * method for potential future implementation
+   * @param partialTemplate
+   * @param context
+   * @param save
+   */
+  static async create(
+    partialTemplate: Partial<IHubTemplate>,
+    context: IArcGISContext,
+    save: boolean = false
+  ): Promise<HubTemplate> {
+    const pojo = this.applyDefaults(partialTemplate, context);
+    const instance = HubTemplate.fromJson(pojo, context);
```

Comment:
Should this throw if we don't want to allow creating them from Hub?

Reply:
I was considering this as well. I think that's a good call. I've updated the underlying `createTemplate` function to throw an error and added comments to the class methods that call that function letting users know we don't currently support creation